src/StreamJsonRpc.Shared/DataContracts/JsonRpcMessage.cs

```diff
@@ -67,7 +67,14 @@ public static JsonRpcMessage CreateRequest(int? id, string @method, IReadOnlyLis
 
         public static JsonRpcMessage CreateRequestWithNamedParameters(int? id, string @method, object namedParameters, JsonSerializer parameterSerializer)
         {
-            return new JsonRpcMessage(method, JToken.FromObject(namedParameters, parameterSerializer), id);
+            if (namedParameters == null)
```

Comment:
Good point on the spec allowing the parameter to be omitted. I thought you were supplying it with a null value with this change and I didn't know it could be omitted entirely.